src/components/icon.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Icon as SUIIcon } from 'semantic-ui-react';
+import { SemanticDatepickerProps } from '../types';
+
+type CustomIconProps = {
+  clearIcon: SemanticDatepickerProps['clearIcon'];
+  icon: SemanticDatepickerProps['icon'];
+  isClearIconVisible: boolean;
+  onClear: () => void;
+  onClick: () => void;
+};
+
+const CustomIcon = ({
+  clearIcon,
+  icon,
```

Comment:
as you are doing a FC, it makes sense to use `defaultProps` since this is a static method.
```suggestion
  clearIcon = 'close',
  icon = 'calendar',
```

Reply:
I don't understand why this would be an improvement. Can you elaborate, please?

The `defaultProps` declaration below seems cleaner for me.

Reply:
read this article, is great: https://blog.logrocket.com/a-complete-guide-to-default-props-in-react-984ea8e6972d/